stan/math/opencl/copy.hpp

```diff
@@ -52,8 +52,9 @@ inline matrix_cl<T> to_matrix_cl(const Eigen::Matrix<T, R, C>& src) {
      */
     cl::Event copy_event;
     const cl::CommandQueue& queue = opencl_context.queue();
-    queue.enqueueWriteBuffer(dst.buffer(), CL_FALSE, 0, sizeof(T) * dst.size(),
-                             src.data(), &dst.write_events(), &copy_event);
```

Comment:
We need to either replace `CL_FALSE` with `CL_TRUE` or setup a separate function in `opencl_context.write_buffer_blocking()`. I dont have a preference.